projects/ngrx.io/content/guide/store/architecture.md

```diff
@@ -1 +1,418 @@
+  products: productsReducer
+})
+</code-example>
+
+When you do that you say that `products` is a thing that will exist in a store and it's state is something we can either *read* or *change*. The way we *read* the value is by subscribing to it using the `Store` service inside of component usually like so:
+
+<code-example header="app.module.ts">
+constructor(private store: Store<AppState>) {
+  this.store.pipe(
+    select('products')
+  ).subscribe(products => console.log('products', products));
+}
+</code-example>
+
+The way we *change* we change our data in the store is by calling the same `Store` service and its `dispatch()` method. The `dispatch()` method will *dispatch*, i.e send an action to the `store` and try to match that to a reducer and specifically a matching case in a reducer. So if we in our component where to do this:
+
+<code-example header="products-list.component.ts">
+add() {
+  this.store.dispatch({ 
+    type: '[Products] add', 
+    { id: 1, name: this.name } 
+  });
+}
+</code-example>
+
+Then we would match the case in our `products.reducer.ts`:
+
+<code-example header="products.reducer.ts">
+case '[Products] add':
+  return [...state, action.payload]
+</code-example>
+
+The end result would be that our `products` store property would now contain one more product.
+
+### When an action fails
+This all seems good right. Set up a property, associate to a reducer, dispatch an action and our state changes. All is good or? All is good UNTIL you mispel your `action.type`. Consider the following code below:
+
+<code-example header="Reducer example">
+function reducer(state = 0, action) {
+  switch(action.type) {
+    case 'INCREMENT':
+      return state + 1;
+    default:
+      return state;
+  }
+}
+// this WON'T work as intended 
+reducer({ type: 'INREMENT' })
+
+// this WILL work, cause it's correctly spelled
+reducer({ type: 'INCREMENT' })
+</code-example>
+
+Above we are showcasing how actions are used, namely that they are objects being sent to a reducer functions that lets them jump through a `switch` construct. We are also demonstrating how the spelling of your action matters. If we spell it `INREMENT` it wont hit the intended case but if we correctly spell it `INCREMENT` it will hit the correct case. 
+
+## The Action interface
+Now we have made the case for why type safety is a good idea. For that reason the `Action` interface exists to make this easy for you when you need to create actions.
+
+The interface is made up of the following:
+
+<code-example header="Action Interface">
+interface Action {
+  type: string;
+}
+</code-example>
+
+The interface has a single property, the `type`, represented as a string. At it's core that's what an action is, an object containing a `type`. The reason the `payload` isn't a property on the `Action` interface is that it's considered *additional context* and is not mandatory. We can definitely add a `payload` property to the class implementing the above interface though.
```

Comment:
>  We can definitely add a `payload` property to the class implementing the above interface though.

Is worth adding? Considering the new `createAction` won't be a class.

Reply:
not sure... I mean ultimately it becomes an object { type: 'message', payload: 'something' }, whether you instantiate a class to accomplish it or createAction()... just so I understand is the Action interface disappearing?